Orange/projection/pca.py

```diff
@@ -137,16 +153,18 @@ def partial_fit(self, data):
         return self
 
 
-class Projector:
-    def __init__(self, projection, feature):
+class Projector(ComputeValueWCommon):
+    """Transform into a given PCA component."""
+
+    def __init__(self, projection, feature, commonfn):
         self.projection = projection
         self.feature = feature
         self.transformed = None
+        self.commonfn = commonfn #domain conversion to original data set,
+                                 #transform to PCA space
 
-    def __call__(self, data):
-        if data.domain != self.projection.pre_domain:
-            data = data.from_table(self.projection.pre_domain, data)
-        return self.projection.transform(data.X)[:, self.feature]
```

Comment:
This could ob course be memoized and shared between different compute_value functions for the same domain. But for lesser memory use that cache would have to be invalidated after transformation. Therefore, I did it so that the common part is exposed to the tranformer, which can destroy the intermediate results after transformation.

I do not get the your first comment, but is probably also irrelevant.